src/lib/variables/variables.ts

```diff
@@ -134,34 +133,42 @@ export const getValue = (
   assertIsVariables(result);
 
   return result;
-};
+}
 
 /**
  * Turns a variable (both primitive and non-primitive) into a string.
  */
-export const variableToString = (variable: Variables): string => {
+export function variableToString(variable: Variables): string {
   if (typeof variable === 'string') {
     return variable;
   }
 
+  if (variable === undefined) {
+    return 'undefined'; // TODO: for both undefined and null, return 'None'?
+  }
+
+  if (Buffer.isBuffer(variable)) {
+    return variable.toString();
+  }
+
   return JSON.stringify(variable);
-};
+}
 
 /**
  * Stringifies a Record of variables. `undefined` values are removed.
  */
-export const variablesToStrings = (
-  variables?: Variables
-): Record<string, string> => {
+export function variablesToStrings(
+  variables: Variables
+): Record<string, string> {
   const result: Record<string, string> = {};
 
-  if (variables !== undefined) {
+  if (!isNone(variables)) {
     for (const [key, value] of Object.entries(variables)) {
-      if (value !== undefined) {
+      if (!isNone(value)) {
```

Comment:
question: What is the purpose of removing the `undefined` from input here? Could this be one of the places where there is an actual difference between `null` and `undefined` in that null is always deliberate while `undefined` could be a sentinel for absence? I don't know how to approach this.

Reply:
I find this function more and more difficult to understand. Even the fact that NonPrimitve's  values are serialized using `JSON.stringify` seem incorrect.

And if I understood original behavior, then null was stringified `'null'`. Discussed on `variableToString`.